server/dns-service/dnsservice-library/src/test/java/org/apache/james/dnsservice/library/netmatcher/NetMatcherTest.java

```diff
@@ -18,95 +18,119 @@
 
-        assertEquals(true, netMatcher.matchInetNetwork("127.0.0.1"));
-        assertEquals(true, netMatcher.matchInetNetwork("localhost"));
-        assertEquals(true, netMatcher.matchInetNetwork("172.16.15.254"));
-        assertEquals(true, netMatcher.matchInetNetwork("192.168.1.254"));
-        assertEquals(false, netMatcher.matchInetNetwork("192.169.1.254"));
+        assertThat(netMatcher.matchInetNetwork("0:0:0:0:0:0:0:1%0")).isFalse();
     }
 
-    /**
-     * @throws UnknownHostException
-     */
     @Test
-    public void testIpV4MatcherWithIpV6() throws UnknownHostException {
+    public void matchInetNetworkShouldMatchWhenIpV6() throws UnknownHostException {
+        netMatcher = new NetMatcher(LOCALHOST_IP_V6_ADDRESSES, DNS_SERVER_IPV6_MOCK);
 
-        netMatcher = new NetMatcher(DNSFixture.LOCALHOST_IP_V4_ADDRESSES, DNSFixture.DNS_SERVER_IPV4_MOCK);
+        assertThat(netMatcher.matchInetNetwork("0:0:0:0:0:0:0:1%0")).isTrue();
+        assertThat(netMatcher.matchInetNetwork("00:00:00:00:00:00:00:1")).isTrue();
+        assertThat(netMatcher.matchInetNetwork("2781:0db8:1234:8612:45ee:ffff:fffe:0001")).isTrue();
+    }
 
-        assertEquals(false, netMatcher.matchInetNetwork("0:0:0:0:0:0:0:1%0"));
-        assertEquals(false, netMatcher.matchInetNetwork("00:00:00:00:00:00:00:1"));
-        assertEquals(false, netMatcher.matchInetNetwork("00:00:00:00:00:00:00:2"));
-        assertEquals(false, netMatcher.matchInetNetwork("2781:0db8:1234:8612:45ee:ffff:fffe:0001"));
-        assertEquals(false, netMatcher.matchInetNetwork("2781:0db8:1235:8612:45ee:ffff:fffe:0001"));
+    @Test
+    public void matchInetNetworkShouldNotMatchWhenIpV6OutOfTheRange() throws UnknownHostException {
+        netMatcher = new NetMatcher(LOCALHOST_IP_V6_ADDRESSES, DNS_SERVER_IPV6_MOCK);
+
+        assertThat(netMatcher.matchInetNetwork("00:00:00:00:00:00:00:2")).isFalse();
+        assertThat(netMatcher.matchInetNetwork("2781:0db8:1235:8612:45ee:ffff:fffe:0001")).isFalse();
+    }
+
+    @Test
+    public void matchInetNetworkShouldNotMatchIpV4WhenIpV6() throws UnknownHostException {
+        netMatcher = new NetMatcher(LOCALHOST_IP_V6_ADDRESSES, DNS_SERVER_IPV6_MOCK);
+
+        assertThat(netMatcher.matchInetNetwork("172.16.15.254")).isFalse();
     }
 
-    /**
-     * @throws UnknownHostException
-     */
     @Test
-    public void testIpV6Matcher() throws UnknownHostException {
+    public void getNetworksShouldNotContainInetNetworkWhenUnknownHostException() {
+        netMatcher = new NetMatcher(ImmutableList.<String>of(HOST_CANNOT_LOOKUP_BY_NAME), DNS_SERVER_IPV6_MOCK);
 
-        netMatcher = new NetMatcher(DNSFixture.LOCALHOST_IP_V6_ADDRESSES, DNSFixture.DNS_SERVER_IPV6_MOCK);
+        assertThat(netMatcher.getNetworks()).isEmpty();
+    }
+    @Test
+    public void getNetworksShouldInitByString() throws Exception {
+        netMatcher = new NetMatcher(IP_V4_IPs, DNS_SERVER_IPV4_MOCK);
```

Comment:
It just to make sure that we can create new NetMatch by the string which contains a list of IP separated by ", ". But I will move it to NetMatcherFactory, is that ok?